### DISCUSSION TITLE
Username change

### DISCUSSION THREAD

**Discussion by marcellovsqz** [community/community#33778] [Other Feature Feedback, Questions, & Ideas] [answered]
If I change my username will I lose all my commit records? Thanks

**Comment by radar-blip** [comment]
This might help, I didn't continue reading all of it, just some of it.

[Changing username - Commits](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username#your-git-commits)

**Reply by marcellovsqz** [reply]
Oh man, this will help me a lot. Thanks

**Reply by radar-blip** [reply]
No problem.

**Comment by Sagar-Sharma-7** [comment, marked as answer]
Hello @marcellomaterializelabs 
After changing your username, your old username becomes available for anyone else to claim. Most references to your repositories under the old username automatically change to the new username. However, some links to your profile won't automatically redirect.
So changing your username will not affect your commits.

**Reply by marcellovsqz** [reply]
Got it man, I'll make the change. Thank you for your help

**Reply by MikeHunt1974** [reply]
Thank you

**Reply by FirstLipKiss** [reply]
Thank you

**Comment by jfullstackdev** [comment]
I did that in the past for 3x, I think so.

it's just always keeping your email because they are linked to commits, even the badges

what you want to watch out for, if you have several repositories, are the links, some will auto-redirect, some won't, that's a tricky one if you share the links in social medias, others will auto-redirect, others wont

another one to be careful is also retirement of your repo name, because if you changed that for a quick period of time, it will return that error, so just don't change it from time to time because it's always part of the URL

**Reply by marcellovsqz** [reply]
Got it man, thanks a lot for the help
